controllers/base_controller.rb

```diff
@@ -0,0 +1,80 @@
+class BaseController
```

Comment:
Your BaseController is not 'base'. 
I mean it knows about game class. ...

Reply:
Yes, because i need it in both Game and Home controllers, that why it here - to avoid code duplication. There only two controllers those might not have initialized controllers: statistics and rules, these controllers have to check game in session before use.